src/utils/conversion.ts

```diff
@@ -4069,7 +4069,26 @@ export function inputStatTypeToMeshStatType(
  * @hidden
  */
 export function meshProposalStatusToProposalStatus(
-  status: PalletMultisigProposalStatus
+  status: PalletMultisigProposalStatus,
+  expiry: Date | null
 ): ProposalStatus {
-  return ProposalStatus[status.type];
+  const { type } = status;
+  switch (type) {
+    case 'ActiveOrExpired':
+      if (!expiry || expiry > new Date()) {
+        return ProposalStatus.Active;
+      } else {
+        return ProposalStatus.Expired;
+      }
+    case 'Invalid':
+      return ProposalStatus.Invalid;
+    case 'ExecutionSuccessful':
+      return ProposalStatus.Successful;
+    case 'ExecutionFailed':
+      return ProposalStatus.Failed;
+    case 'Rejected':
+      return ProposalStatus.Rejected;
+    default:
+      throw new UnreachableCaseError(type);
```

Comment:
```suggestion
       throw new PolymeshError({
    code: ErrorCode.UnexpectedError,
    message: 'Unsupported Multisig Proposal Status. Please contact the Polymesh team',
    data: {
      auth: JSON.stringify(status, null, 2),
    },
  });
```

Reply:
The `UnreachableCaseError` takes an argument of type `never`. By using it we ensure all cases are checked, so if a new one is added we will get a compile error.

We still throw in case someone is using JavaScript, but in TypeScript, the compiler ensures it won't get triggered.

This is also why I checked the strings, but using the `isActiveOrExpired` doesn't narrow the type down of status, where as using the `type` will.